internal/descriptor/provider/provider.go

```diff
@@ -1,109 +1,90 @@
 package provider
 
 import (
+	"context"
 	"errors"
 	"fmt"
 
-	"ocm.software/ocm/api/ocm/compdesc"
-
-	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
 	"github.com/kyma-project/lifecycle-manager/internal/descriptor/cache"
 	"github.com/kyma-project/lifecycle-manager/internal/descriptor/types"
+	"github.com/kyma-project/lifecycle-manager/internal/descriptor/types/ocmidentity"
 )
 
 var (
-	ErrTypeAssert    = errors.New("failed to convert to v1beta2.Descriptor")
-	ErrDecode        = errors.New("failed to decode to descriptor target")
-	ErrTemplateNil   = errors.New("module template is nil")
-	ErrDescriptorNil = errors.New("module template contains nil descriptor")
+	ErrDecode             = errors.New("failed to decode to descriptor target")
+	ErrNilProvider        = errors.New("OCMIProvider is nil")
+	ErrNilIdentity        = errors.New("component identity is nil")
+	ErrNameOrVersionEmpty = errors.New("component name or version is empty")
 )
 
+type DescriptorService interface {
+	GetComponentDescriptor(ctx context.Context, ocmi ocmidentity.Component) (*types.Descriptor, error)
+}
+
 type CachedDescriptorProvider struct {
-	DescriptorCache *cache.DescriptorCache
+	descriptorCache   *cache.DescriptorCache
+	descriptorService DescriptorService
 }
 
-func NewCachedDescriptorProvider() *CachedDescriptorProvider {
+func NewCachedDescriptorProvider(service DescriptorService) *CachedDescriptorProvider {
 	return &CachedDescriptorProvider{
-		DescriptorCache: cache.NewDescriptorCache(),
+		descriptorCache:   cache.NewDescriptorCache(),
+		descriptorService: service,
 	}
 }
 
-func (c *CachedDescriptorProvider) GetDescriptor(template *v1beta2.ModuleTemplate) (*types.Descriptor, error) {
-	if template == nil {
-		return nil, ErrTemplateNil
-	}
-
-	if template.Spec.Descriptor.Object != nil {
-		desc, ok := template.Spec.Descriptor.Object.(*types.Descriptor)
-		if !ok {
-			return nil, ErrTypeAssert
-		}
+// Convenience interface to get the OCM identity of a component from objects
+// that already have all required data.
+// Then we don't have to create intermediate variables of type ocmidentity.Component.
+type OCMIProvider interface {
+	GetOCMIdentity() (*ocmidentity.Component, error)
+}
```

Comment:
Wondering if the `ModuleTemplateInfo` could also directly implement the `ocmidentity.Component` interface (`Name()`, `Version()`). Then we wouldn't need this interface and could directly pass the `ModuleTemplateInfo`. But the function names are rather generic, so also not too sure if this is a good idea.

But if we keep the interface, could we move the type up to before the constructor? I think it is easier to read if the file is ordered like "types -> vars -> constructor -> funcs"